docs/ProfilingDevelopment.md

```diff
@@ -0,0 +1,91 @@
+# Profiling Development
+
+## What?
+
+This file contains development notes specific to the profiling feature.
+
+For a more practical view of getting started with development of `ddtrace`, see <DevelopmentGuide.md>.
+
+## Profiling components high-level view
+
+Components below live inside <../lib/ddtrace/profiling>:
+
+* `Collectors::Stack`: Collects stack trace samples from Ruby threads for both CPU-time (if available) and wall-clock.
+  Runs on its own background thread.
+* `Encoding::Profiler`: Encodes gathered data into the pprof format.
+* `Events::Stack`, `Events::StackSample`, `Events::StackExceptionSample`: Entity classes used to represent stacks.
+* `Ext::CPU`: Monkey patches Ruby's `Thread` with our `Ext::CThread` to enable CPU-time profiling.
+* `Ext::CThread`: Extension used to enable CPU-time profiling via use of Pthread's `getcpuclockid`.
+* `Ext::Forking`: Monkey patches `Kernel#fork`, adding a `Kernel#at_fork` callback mechanism which is used to restore
+  profiling abilities after the VM forks (re-instrument main thread, restart profiler threads).
+* `Pprof::*` (in <../lib/ddtrace/profiling/pprof>): Converts samples captured in the `Recorder` into the pprof format.
+* `Tasks::Setup`: Takes care of loading our extensions/monkey patches and starting up profiler. See initialization
+  section below for more details.
+* `Transport::*` (in <../lib/ddtrace/profiling/transport>): Implements reporting of pprof payloads to the datadog agent
```

Comment:
I would say this is more of `Implements transmission of profiling payloads to Datadog agent`, as reporting is more associated with analytics, and pprof is a detail.